risc0/zkvm/sdk/rust/host/src/lib.rs

```diff
@@ -152,6 +161,32 @@ impl Receipt {
     }
 }
 
+impl Serialize for Receipt {
+    /// Generate a serialized version of the whole receipt.
+    fn serialize<S: Serializer>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error> {
+        let mut seq = serializer.serialize_tuple(2)?;
+        seq.serialize_element(self.get_journal().unwrap())?;
+        seq.serialize_element(self.get_seal().unwrap())?;
+        seq.end()
+    }
+}
+
+impl<'de> Deserialize<'de> for Receipt {
+    /// Deserialize a receipt.
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct ReceiptData {
+            journal: Vec<u8>,
+            seal: Vec<u32>,
+        }
```

Comment:
Maybe we could use this for both serialize and deserialize to ensure both have the same exact encoding, along with the ability to use the serde derive attributes.